obp/ope/estimators_multi.py

```diff
@@ -0,0 +1,1806 @@
+        position: Optional[np.ndarray] = None,
+        **kwargs,
+    ) -> np.ndarray:
+        """Estimate round-wise (or sample-wise) rewards.
+
+        Parameters
+        ----------
+        reward: array-like, shape (n_rounds,)
+            Rewards observed for each data in logged bandit data, i.e., :math:`r_i`.
+
+        action: array-like, shape (n_rounds,)
+            Actions sampled by the logging/behavior policy for each data in logged bandit data, i.e., :math:`a_i`.
+
+        pscore: array-like, shape (n_rounds,)
+            Action choice probabilities of the logging/behavior policy (propensity scores), i.e., :math:`\\pi_k(a_i|x_i)`.
+            If `use_estimated_pscore` is False, `pscore` must be given.
+
+        stratum_idx: array-like, shape (n_rounds,)
+            Indices to differentiate the logging/behavior policy that generate each data, i.e., :math:`k`.
+
+        action_dist: array-like, shape (n_rounds, n_actions, len_list)
+            Action choice probabilities of the evaluation policy (can be deterministic), i.e., :math:`\\pi_e(a_i|x_i)`.
+
+        estimated_rewards_by_reg_model: array-like, shape (n_rounds, n_actions, len_list)
+            Estimated expected rewards given context, action, and position, i.e., :math:`\\hat{q}(x_i,a_i)`.
+
+        position: array-like, shape (n_rounds,), default=None
+            Indices to differentiate positions in a recommendation interface where the actions are presented.
+            If None, the effect of position on the reward will be ignored.
+            (If only a single action is chosen for each data, you can just ignore this argument.)
+
+        Returns
+        ----------
+        estimated_rewards: array-like, shape (n_rounds,)
+            Estimated rewards for each observation.
+
+        """
+        if position is None:
+            position = np.zeros(action_dist.shape[0], dtype=int)
+        iw = action_dist[np.arange(action.shape[0]), action, position] / pscore
+        # weight clipping
+        if isinstance(iw, np.ndarray):
+            iw = np.minimum(iw, self.lambda_)
+
+        n = action.shape[0]
+        q_hat_at_position = estimated_rewards_by_reg_model[np.arange(n), :, position]
+        q_hat_factual = estimated_rewards_by_reg_model[np.arange(n), action, position]
+        pi_e_at_position = action_dist[np.arange(n), :, position]
+        estimated_rewards = np.average(
+            q_hat_at_position,
+            weights=pi_e_at_position,
+            axis=1,
+        )
+
+        unique_stratum_idx, n_data_strata = np.unique(stratum_idx, return_counts=True)
+        var_k = np.zeros(unique_stratum_idx.shape[0])
+        for k in unique_stratum_idx:
+            idx_ = stratum_idx == k
+            var_k[k] = np.var(
+                estimated_rewards[idx_]
+                + iw[idx_] * (reward[idx_] - q_hat_factual[idx_])
+            )
+        weight_k = n / (var_k * np.sum(n_data_strata / var_k))
```

Comment:
[ask]
Is this a typo?

```suggestion
        weight_k = n_data_strata / (var_k * np.sum(n_data_strata / var_k))
```